### DISCUSSION TITLE
Watchtower implementation

### DISCUSSION THREAD

**Discussion by gcool-info** [ACINQ/eclair#2590] [Node operators]
Hey there! I was wondering if `eclair` can be used with a watchtower?

**Comment by t-bast** [comment]
Not currently, because watchtowers don't really make sense without Eltoo and eclair generally doesn't need it.
Why do you think you need a watchtower? What's your use-case for it?

**Reply by gcool-info** [reply]
Thanks @t-bast 

Suppose your lightning node goes down.  During that time, a channel counterparty broadcasts a revoked state to exploit your node's downtime. As I understand it, a watchtower's purpose is to keep tabs and broadcast a justice transaction, to safeguard against this.

Does `eclair` work in a different way?

**Reply by t-bast** [reply]
There are some common misconceptions here, you very likely don't need a watchtower if this is what you're afraid of.

A channel counterparty that performs this attack would be taking a very high risk: the csv delays to actually claim funds from their revoked state are quite long (720 blocks by default in `eclair`). If your node comes back online before those 720 blocks, your channel counterparty will be the one losing money (at least their channel reserve and the on-chain fees for channel transactions). If you have money at stake in a lightning node, you should really have decent enough ops/monitoring to detect that your node is down and restart it before 5 days (which is configurable, you could set it to a higher value if you wish) have elapsed (and your node has no good reason to crash in the first place). If not, why would your watchtower be more resilient than your lightning node and actually save you?

If the peer attacking you has taken control of the machine where you run your lightning node, they will most likely also be able to take control of the machine where you run the watchtower. Or they will simply block outgoing traffic to the watchtower, making it useless. If that happens, watchtowers don't save you either.

To summarize, running a watchtower is additional complexity which doesn't provide strong guarantees for server nodes. It could make sense for mobile wallets, but only in the future with Eltoo once watchtowers aren't prohibitively expensive to run (and even then, I'm not convinced we'll see a lot of usage).

**Reply by gcool-info** [reply]
Thanks, makes sense :)
